kasa/smartdevice.py

```diff
@@ -760,7 +783,7 @@ def internal_state(self) -> Any:
         The returned object contains the raw results from the last update call.
         This should only be used for debugging purposes.
         """
-        return self._last_update
+        return self._last_update or self._discovery_info
```

Comment:
How about setting `_last_update` to the discovery info whenever it is assigned? This would avoid doing this altogether.

Reply:
At the moment that would break the way the `requires_update` is currently implemented for devices `update_from_discovert_info`.  We may want to look at that implementation when we introduce the interfaces.